selfdrive/controls/lib/longitudinal_planner.py

```diff
@@ -93,11 +97,20 @@ def parse_model(model_msg, model_error):
       v = np.zeros(len(T_IDXS_MPC))
       a = np.zeros(len(T_IDXS_MPC))
       j = np.zeros(len(T_IDXS_MPC))
-    return x, v, a, j
+    if len(model_msg.meta.disengagePredictions.gasPressProbs) > 1:
+      throttle_prob = model_msg.meta.disengagePredictions.gasPressProbs[1]
+    else:
+      throttle_prob = 1.0
+    return x, v, a, j, throttle_prob
 
   def update(self, sm):
     self.mpc.mode = 'blended' if sm['selfdriveState'].experimentalMode else 'acc'
 
+    if len(sm['carControl'].orientationNED) == 3:
+      accel_coast = np.sin(sm['carControl'].orientationNED[1]) * -5.65 - 0.3  # fitted from data
```

Comment:
This should be a function called get_coast_accel. And there should be a comment that says with which command these values were generate in xx.

Reply:
Where does the np.sin come from? Is the linear fit better with the sin?

Reply:
The sine makes almost no difference, it only starts diverges from linear at extreme values (>30deg) where we have relatively little data. It's there because it seemed more correct to me, in a frictionless gravitational model you'd expect acceleration to be sin(pitch) * -9.81 m/s^2.